lib/Installation/ModuleRegistration/ModuleRegCodePage.pm

```diff
@@ -0,0 +1,40 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: The module provides interface to act with page that ask for module extra registration code
+#
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+package Installation::ModuleRegistration::ModuleRegCodePage;
+use parent 'Installation::Navigation::NavigationBase';
+use strict;
+use warnings;
+
+sub new {
+    my ($class, $args) = @_;
+    my $self = bless {
+        app => $args->{app}
+    }, $class;
+    return $self->init();
+}
+
+sub init {
+    my ($self) = @_;
+    $self->SUPER::init();
+    $self->{tb_we_code} = $self->{app}->textbox({id => '"sle-we"'});
+    return $self;
+}
+
+sub is_shown {
+    my ($self) = @_;
+    return $self->{tb_we_code}->exist();
```

Comment:
could we use a generic id, so this page can be reused for another module with additional registration.
https://openqa.suse.de/tests/9359062#step/register_extensions_and_modules/2
the extension you selected ... might do.